### PR TITLE
test(ai-hub): add MCP registry integration test

### PR DESCRIPTION
**Pull request by jonburdo**
Add backend integration coverage for registering an MCP Catalog entry in MLflow MCP Registry, deploying it through the dashboard BFF APIs, and verifying the resulting MCP endpoint is reachable on-cluster.

# Pull Request

## Summary

This adds a test to ensure we can register from MCP Catalog to MCP Registry and then deploy the mcp server.

Tested by running locally:
```
uv run --frozen pytest tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py -vv -s --cluster-sanity-skip-check
```
after `oc login` on https://rh-ai.apps.rosa.jburdo2.53ms.p3.openshiftapps.com/ with RHOAI nightly 3.6 ea1 installed:
`quay.io/rhoai/rhoai-fbc-fragment:rhoai-3.6-ea.1@sha256:90ed438388cbed021f78f2634b35ee68f6019b64e7cb91bed92d5aba49ee6b8f`

## Related Issues

- Fixes: 
- JIRA: 

## Please review and indicate how it has been tested

- [x] Locally
- [ ] Jenkins

## Additional Requirements

- [ ] If this PR introduces a new test image, did you create a PR to mirror it in disconnected environment?
- [ ] If this PR introduces new marker(s)/adds a new component, was relevant ticket created to update relevant Jenkins job?

### CODE REVIEW

File: tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+                    "displayName": "QE MCP registry deployment",
+                    "registryServer": server_name,
+                    "registryVersion": DEPLOYABLE_MCP_VERSION,
+                    "image": deploy_spec["source"]["containerImage"]["ref"],
+                    "yaml": deploy_yaml_from_spec(deploy_spec=deploy_spec),
+                }
+            },
+        )
+        deployment_data = extract_data_envelope(payload=deployment_payload, url=deployment_url)
+        deployment_name = deployment_data["name"]
```
Comment (dkuc) [inline tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py]: **Resource leak: cleanup flags/names captured only after `extract_data_envelope`**

The register POST (line 145) and deployment POST (line 165) can return `201` — meaning the resource is genuinely created — but if `extract_data_envelope` then raises `AssertionError` (envelope lacks a dict `data`), `created_registry_version` stays `False` / `deployment_name` stays `None`. The `finally` block then skips deletion, orphaning the registry version and MCP deployment on the cluster.

Set the cleanup flag/name *before* parsing the response envelope, so teardown runs whenever the POST returned 201.

File: tests/ai_hub/mcp_servers/integration/utils.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        expected_status_codes={200},
+    )
+
+    last_seen: dict[str, Any] = {}
+    for sample in sampler:
+        last_seen = extract_data_envelope(payload=sample, url=deployment_url)
+        if deployment_ready(deployment=last_seen):
+            return last_seen
+
+    pytest.fail(f"MCP deployment {deployment_name} did not become Ready: {last_seen}")
```
Comment (dkuc) [inline tests/ai_hub/mcp_servers/integration/utils.py:166]: **Unreachable `pytest.fail` on deployment timeout**

`TimeoutSampler` raises `TimeoutExpiredError` at timeout rather than ending the `for sample in sampler` loop via `StopIteration`. As a result, this `pytest.fail(...)` with `last_seen` never runs — a timeout surfaces as a generic `TimeoutExpiredError` with no captured deployment state.

Wrap the sampler loop in `try/except TimeoutExpiredError` (or restructure) so the diagnostic with `last_seen` actually fires on timeout.

**Review by dkuc — APPROVED** [review]

**Comment by dkuc** [comment]
Posted a couple edge case findings - I'm fine skipping them - up to you.  Otherwise, Looks good! I was able to successfully run the test.  Approved.

File: utilities/resources/mcp_server.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Generated using https://github.com/RedHatQE/openshift-python-wrapper/blob/main/scripts/resource/README.md
```
Comment (lugi0) [inline utilities/resources/mcp_server.py:1]: this would need to go to the python wrapper

File: tests/ai_hub/mcp_servers/integration/conftest.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    DEPLOYABLE_MCP_VERSION,
+)
+from tests.ai_hub.utils import execute_get_command_with_retry, wait_for_catalog_api
+from utilities.resources.pod import Pod as UtilPod
+
+LOGGER = structlog.get_logger(name=__name__)
+
+
+@pytest.fixture(scope="class")
+def mcp_registry_test_metadata(model_namespace: Namespace) -> dict[str, Any]:
```
Comment (lugi0) [inline tests/ai_hub/mcp_servers/integration/conftest.py:142]: I think most of these variables could be defined as constants in the relevant file and then you'd only append the suffix as needed here

File: tests/ai_hub/mcp_servers/integration/conftest.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        "server_name": server_name,
+        "source_id": source_id,
+        "source_name": source_name,
+        "yaml_catalog_path": yaml_catalog_path,
+        "catalog_yaml": catalog_yaml,
+    }
+
+
+@pytest.fixture(scope="class")
+def dashboard_api_base_url(dashboard_pod: UtilPod) -> Generator[str, Any]:
```
Comment (lugi0) [inline tests/ai_hub/mcp_servers/integration/conftest.py:225]: why do you need to go through the BFF?

File: tests/ai_hub/mcp_servers/integration/conftest.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    ).get("size", 0)
+
+    with ResourceEditor(patches={catalog_config_map: patches}):
+        wait_for_catalog_api(
+            endpoint="mcp_servers",
+            item_name="MCP servers",
+            url=mcp_catalog_rest_urls[0],
+            headers=model_registry_rest_headers,
+            previous_size=pre_patch_size,
+            expected_size=pre_patch_size + 1,
```
Comment (lugi0) [inline tests/ai_hub/mcp_servers/integration/conftest.py:291]: do you only want to check that it has increased in size or also that the entry is correct?

File: tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+    mcp_catalog_rest_urls: list[str],
+    mcp_registry_test_metadata: dict[str, Any],
+    model_namespace: Namespace,
+    model_registry_rest_headers: dict[str, str],
+    unused_tcp_port_factory: Callable[[], int],
+) -> None:
+    """
+    Given: A deployable MCP server exists in the live AI Hub MCP catalog
+    When: Its catalog metadata is registered in MLflow and the registered version is deployed through the
+          dashboard BFF APIs
```
Comment (lugi0) [inline tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py:41]: same as above, do you need to go through the BFF? It's unusual for our tests

File: tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+    [
+        pytest.param(
+            {"name": "test-mcp-registry-happy-path"},
+            marks=[pytest.mark.install, pytest.mark.tier2],
+            id="test_mcp_registry_backend_happy_path",
+        )
+    ],
+    indirect=True,
+)
+def test_register_and_deploy_mcp_server_from_catalog_backend_happy_path(
```
Comment (lugi0) [inline tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py:32]: This test is very long and has a lot of asserts compared to our usual style. 
Would it be possible to split this into multiple tests, or alternatively to abstract away some of the logic in fixtures? 

If I understand correctly everything before the try is setup and sanity checks on the running resources, and if so it would be ideal to have these steps in fixtures rather than in the test. 

The `finally` block is also a bit unusual, we'd expect teardown to be performed by whatever fixture first created the resource

File: tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py (hunk 1 above)
Comment (jonburdo) [reply tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py]: Good catch, fixed

File: tests/ai_hub/mcp_servers/integration/utils.py (hunk 2 above)
Comment (jonburdo) [reply tests/ai_hub/mcp_servers/integration/utils.py:166]: Fixed

File: tests/ai_hub/mcp_servers/integration/conftest.py (hunk 5 above)
Comment (jonburdo) [reply tests/ai_hub/mcp_servers/integration/conftest.py:225]: The goal of the test is to cover the same dashboard-driven register/deploy flow the product uses, so it goes through the dashboard BFF rather than bypassing it with lower-level direct calls

File: tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py (hunk 8 above)
Comment (jonburdo) [reply tests/ai_hub/mcp_servers/integration/test_mcp_registry_happy_path.py:32]: I agreed with the fixture/teardown part and refactored most of the orchestration into fixtures so the test body is much smaller and resource cleanup is owned closer to creation. I kept it as one test, though, because the catalog->registry->deploy->endpoint path is the single end-to-end behavior this PR is trying to validate.

File: tests/ai_hub/mcp_servers/integration/conftest.py (hunk 6 above)
Comment (jonburdo) [reply tests/ai_hub/mcp_servers/integration/conftest.py:291]: Good point. This size check tells us the catalog API has picked up the source update. I’ve also added a dedicated `catalog_server_details` fixture that resolves the exact injected entry by `name` and `source_id` and validates its detail/tools payload before the test uses it

**Review by dkuc — APPROVED** [review]

**Review by lugi0 — APPROVED** [review]

File: utilities/resources/mcp_server.py (hunk 3 above)
Comment (jonburdo) [reply utilities/resources/mcp_server.py:1]: Created a PR here: https://github.com/RedHatQE/openshift-python-wrapper/pull/2796
